src/host/PtySignalInputThread.cpp

```diff
@@ -272,26 +287,21 @@ void PtySignalInputThread::_DoSetWindowParent(const SetParentData& data)
 bool PtySignalInputThread::_GetData(_Out_writes_bytes_(cbBuffer) void* const pBuffer,
                                     const DWORD cbBuffer)
 {
+    if (!_hFile)
+    {
+        return false;
+    }
+
     DWORD dwRead = 0;
-    // If we failed to read because the terminal broke our pipe (usually due
-    //      to dying itself), close gracefully with ERROR_BROKEN_PIPE.
-    // Otherwise throw an exception. ERROR_BROKEN_PIPE is the only case that
-    //       we want to gracefully close in.
     if (FALSE == ReadFile(_hFile.get(), pBuffer, cbBuffer, &dwRead, nullptr))
     {
-        auto lastError = GetLastError();
-        if (lastError == ERROR_BROKEN_PIPE)
-        {
-            _Shutdown();
-        }
-        else
-        {
-            THROW_WIN32(lastError);

```

Comment:
Regarding...
```cpp
        auto lastError = GetLastError();
        if (lastError == ERROR_BROKEN_PIPE)
        {
            _Shutdown();
        }
        else
        {
            THROW_WIN32(lastError);
        }
```
Looks like we never really handled the thrown error, right? So I guess it's fine that we removed it. But should we at least collect and log it?